### PR TITLE
Help framework

### PR DESCRIPTION
**Pull request by natoverse**
- Adds core API methods for profiles to supply a help content index and for components to indicate the current context-sensitive help to display.
- Adds a guidance panel to the ResourcePane with an expand/collapse help guidance section.
- Updates the core docs to provide some help content for each of the built-in resource types.

### CODE REVIEW

File: docs/resources/codebook.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+## Schema values
+
+Each column has attributes you can edit that affect how the data is interpreted and displayed. Column schemas can be quite complex, and you can find full details in our [API documentaton](https://github.com/microsoft/datashaper/blob/main/javascript/schema/docs/markdown/schema.field.md). The most important attributes for most uses are shown below.
+
+- Display name: this is a user-friendly name we should show throughout the application, because data column names are often cryptic.
+- Description: this provides a place for detailed descriptions of what the column represents.
+- Data type: the data type tells processing algorithms how to interpret the data in a column. For example "hello" should be a string, while "1,300" may be treated as a number for computations. If a cell value in the column cannot be interpreted as the stated data type, it will be treated as empty (`null`).
+- Data nature: this defines the semantic shape of column data values. For example, if a column consists only of 5-10 unique strings, this is probable "categocial" data, and should be treated differently than "continuous" numeric data.
```
Comment (darthtrevino) [inline docs/resources/codebook.md:12]: categocial => categorical

File: docs/resources/codebook.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Codebook
+
+A **codebook** defines the schema for [data table](./datatable.md) columns. Each column can define such attributes as the core data type, data nature, and units.
+
+## Schema values
+
+Each column has attributes you can edit that affect how the data is interpreted and displayed. Column schemas can be quite complex, and you can find full details in our [API documentaton](https://github.com/microsoft/datashaper/blob/main/javascript/schema/docs/markdown/schema.field.md). The most important attributes for most uses are shown below.
+
+- Display name: this is a user-friendly name we should show throughout the application, because data column names are often cryptic.
```
Comment (darthtrevino) [inline docs/resources/codebook.md:9]: I think args should be either italicized or bolded, but that's my own pref

File: javascript/app-framework/src/components/app/DataShaperApp/DataShaperApp.tsx [hunk 3]
```diff
@@ -112,7 +118,7 @@ const AppInner: React.FC<DataShaperAppProps> = memo(function AppInner({
 								<Route
 									key={r.href}
 									path={(r.children?.length ?? 0) > 0 ? r.href : `${r.href}/*`}
-									element={<MatchedRoute key={r.href} data={r} />}
+									element={<MatchedRoute key={r.href} data={r} api={api} />}
```
Comment (darthtrevino) [inline javascript/app-framework/src/components/app/DataShaperApp/DataShaperApp.tsx:121]: Interesting - a context value may make sense as well (in the future)

**Review by darthtrevino — APPROVED** [review]

File: javascript/app-framework/src/components/app/DataShaperApp/DataShaperApp.tsx (hunk 3 above)
Comment (natoverse) [reply javascript/app-framework/src/components/app/DataShaperApp/DataShaperApp.tsx:121]: I had a similar thought, but it also seemed reasonable that the parent profile renderers would just get it as a prop. Could go either way.

**Pull request merged**